### DISCUSSION TITLE
add -j buildflag as argument to bootstrap-vcpkg

### DISCUSSION THREAD

**Discussion by L0ric0** [microsoft/vcpkg#22754] [Q&A]
****

**Is your feature request related to a problem? Please describe.**
When building vcpkg on systems that have low memory compared to the number of cpu cores bootstraping takes several times longer due to swapping or fails due to OOM. This is due to the fact that the bootstraping process always spawns the maximum number of processes to build vcpkg.

**Proposed solution**
Adding a -j flag (which takes a number setting the number of parralell build processes) to the `bootstrap-vcpkg.{sh,bat}` scripts which is forwarded to cmake should resove this problem for building by directly calling the scripts but also for manifest where it can be added to `VCPKG_BOOTSTRAP_OPTIONS`.

**Describe alternatives you've considered**
The same can be achived by setting a cmake environment variable but this would also affect further cmake calles in manifest mode and is sometime hard to set (visual studio remote builds)

**Additional context**
Building vcpkg on a raspberry pi 3 spawns 6 g++ processes which consume 2 - 2.5 GB of ram at peak the pi3 has only 1GB of ram and building with the 6 processes requires around 3h where running the build with only one g++ process (by editing `scripts/bootstrap.sh`) finishes in around 1h.

**Comment by dg0yt** [comment]
You can set environment variable `VCPKG_MAX_CONCURRENCY` to the desired limit of parallel jobs. It is the same variable which is used when building ports.
